app/views/tasks/show.html.slim

```diff
@@ -574,9 +574,13 @@
             =< t('common.button.duplicate')
 
       - if policy(@task).generate_test?
```

Comment:
I know you discussed about this line already... I am still wondering why you are making `generate_test?` a synonym of `update?` effectively. Wouldn't it make sense to have both, the OpenAI API key check and the check for `update?` combined? Then, we would need two cases here (pseudo-code):

```
- if generate_test?
  # show enabled button
- elsif edit?
  # show disabled button
- else
  # show no button // no op
```

Reply:
there is no option to not show the button at all. the button will be visible regardless.

I added the check for the openAI key but @kkoehn removed it in the last commit. but the functionality is still working as intended. not sure if its then really needed to have that check

Reply:
Karol mentioned that the API key check is done independently from here. https://github.com/openHPI/codeharbor/pull/1496#discussion_r1671268295